core-api/src/main/java/com/optimizely/ab/bucketing/Bucketer.java

```diff
@@ -102,6 +109,19 @@ private Variation bucketToVariation(@Nonnull Experiment experiment,
                                         @Nonnull String userId) {
         // "salt" the bucket id using the experiment id
         String combinedBucketId = userId + experiment.getId();
+
+        // If a persistent bucketer instance is present then check it for a persisted variation
+        if (userExperimentRecord != null) {
+            String variationKey = userExperimentRecord.lookup(userId, experiment.getKey());
+            if (variationKey != null) {
+                logger.info("Returning previously activated variation \"{}\" from user experiment registry.", variationKey);
```

Comment:
Also, the log comment needs to be updated. It is not coming from registry any more.